### DISCUSSION TITLE
How to search by product variant?

### DISCUSSION THREAD

**Discussion by junereycasuga** [vendurehq/vendure#1283] [Q&A]
I am setting up an e-commerce website for selling clothes

I have created the product and added variants to it by colors and sizes. How do I filter products by color and sizes? e.g. I would like to get all the clothes with size small or color yellow

**Comment by michaelbromley** [comment]
There's no way to filter by the options at the monent. You could achieve this by using the [ElasticsearchPlugin ](https://www.vendure.io/docs/typescript-api/elasticsearch-plugin/)and making us of the customMappings and mapQuery options.
